### PR TITLE
doc_reorg for diataxis

### PR DESCRIPTION
**Pull request by randyh62**
resolves #___

Summary of proposed changes:
- updated index.rst for new card based TOC
- edited content for consistency
-

### CODE REVIEW

File: docs/functions.rst [hunk 1]
```diff
@@ -57,40 +60,44 @@ however functions which support this alternate API include the line:
...
+For simplified usage with Hipified code, there is an option to interpret the API as using ``hipComplex`` and ``hipDoubleComplex``
+types (i.e. ``typedef hipComplex hipblasComplex``). This is provided for users to avoid casting when using the hip complex types in their code.
+As the memory layout is consistent across all three types, it is safe to cast arguments to API calls between the 3 types: ``hipComplex``,
+``std::complex<float>``, and ``hipblasComplex``, as well as for the double precision variants. To expose the API as using the hip defined complex types,
+users can use either a compiler define or inline ``#define ROCM_MATHLIBS_API_USE_HIP_COMPLEX`` before including the header file `<hipblas.h>`. Thus, the
 API is compatible with both forms, but recompilation is required to avoid casting if switching to pass in the hip complex types.
 
-Note that hipblasComplex, hipblasDoubleComplex, and use of ROCM_MATHLIBS_API_USE_HIP_COMPLEX are now deprecated. The API will provide interfaces
-using only hipComplex and hipDoubleComplex in the future.
+.. note::
```
Comment (daineAMD) [inline docs/functions.rst:101]: It should also be noted here that when using the `HIPBLAS_V2` define (as explained above), the API will be exposed with the `hipComplex` and `hipDoubleComplex` datatypes. You can add that here or I can in another PR if you'd rather.

File: docs/index.rst [hunk 2]
```diff
@@ -1,12 +1,38 @@
...
+
+    * :ref:`hipblas-orga`
+    * :ref:`hipblas-clients`
+    * :ref:`contribute`
+
+  .. grid-item-card:: Reference
+
+    * :ref:`api_label`
+    * :ref:`deprecations`
+
```
Comment (daineAMD) [inline docs/index.rst:34]: This doesn't display as intended in the "rich diff" in github. Might just be a github limitation, but make sure it displays correctly in the build for you.

**Review by daineAMD — COMMENTED** [review]
Looks like the [docs build is failing](https://readthedocs.com/projects/advanced-micro-devices-hipblas/builds/1973721/), not sure if you were aware.

File: docs/functions.rst (hunk 1 above)
Comment (TorreZuk) [reply docs/functions.rst:101]: > It should also be noted here that when using the `HIPBLAS_V2` define (as explained above), the API will be exposed with the `hipComplex` and `hipDoubleComplex` datatypes. You can add that here or I can in another PR if you'd rather.

I didn't know if hipComplex is preferred over hipFloatComplex?  If so can we only use that form?

Comment (daineAMD) [reply docs/functions.rst:101]: I suggested opting for hipComplex. I see them both in the HIP documentation so not sure which is preferred. I would assume anyone porting from cuda would expect hipComplex over hipFloatComplex, though.

File: docs/index.rst (hunk 2 above)
Comment (randyh62) [reply docs/index.rst:34]: It looks correct in the generated HTML page.

**Comment by randyh62** [comment]
> Looks like the [docs build is failing](https://readthedocs.com/projects/advanced-micro-devices-hipblas/builds/1973721/), not sure if you were aware.

I was not aware. I fixed it.

**Comment by randyh62** [comment]
I have merged Daine's changes, and fixed the build error. Should be good to go now.

File: docs/functions.rst [hunk 3]
```diff
@@ -59,43 +62,46 @@ however functions which support this alternate API include the line:
...
-hipComplex and hipDoubleComplex.
+``hipblasComplex`` and ``hipblasDoubleComplex`` are also deprecated. In a future release, all uses of these types will be replaced with their HIP counterparts:
+``hipComplex`` and ``hipDoubleComplex``.
 
-While hipblasDatatype_t, ``hipblasComplex``, and ``hipblasDoubleComplex`` are deprecated, users may use the compiler define or inline #define HIPBLAS_V2 before including the header file <hipblas.h> to
-access the updated API. In a future release, this define will no longer be needed and deprecated functions will be removed, leaving the updated interface.
+While ``hipblasDatatype_t``, ``hipblasComplex``, and ``hipblasDoubleComplex`` are deprecated, users may use the compiler define or inline ``#define HIPBLAS_V2`` before including the header file `<hipblas.h>` to access the updated API. In a future release, this define will no longer be needed and deprecated functions will be removed, leaving the updated interface.
 Please see the documentation for the following functions to see the new interfaces using hipDataType: hipblasTrsmEx, hipblasGemmEx,
```
Comment (daineAMD) [inline docs/functions.rst]: Sorry missed this, on L73 you might want the double `` around hipDataType, and maybe the function names.

**Review by daineAMD — APPROVED** [review]

File: docs/functions.rst (hunk 3 above)
Comment (randyh62) [reply docs/functions.rst]: Fixed.

**Pull request merged**